dot-net-sdk/dto/AssignmentLogData.cs

```diff
@@ -3,13 +3,23 @@ namespace eppo_sdk.dto;
 public class AssignmentLogData
 {
     public string experiment;
+    public string feature_flag;
+    public string allocation;
     public string variation;
     public DateTime timestamp;
     public string subject;
     public SubjectAttributes subjectAttributes;
 
-    public AssignmentLogData(string experiment, string variation, string subject, SubjectAttributes subjectAttributes) {
-        this.experiment = experiment;
+    public AssignmentLogData(
+        string feature_flag,
+        string allocation,
+        string variation,
+        string subject,
+        SubjectAttributes subjectAttributes)
+    {
+        this.experiment = feature_flag + "-" + allocation;
```

Comment:
I'm not sure we decided _how_ to handle it, but do we want to handle the case of a null allocation? ([per this comment thread](https://www.notion.so/Concatenated-experiment-key-design-d212aaaf42e7440bbdafd13cd0069dc8?d=900bdf54a4ac45cab39ac7b369772553&pvs=4#24eb8b56a36f4defbdf07eef352b3e28))

Reply:
Also, in the Java SDK we are [base64 encoding the allocation](https://github.com/Eppo-exp/java-server-sdk/pull/20/files#diff-332b1a069e8814fb4b0c8f34e18078df1bafc1a9150890149f31181a170740dbR10)--is this something we want to do consistently across all SDKs?

Reply:
Yep it's a bit inconsistent. Sameeran made a decision in a comment thread about how to handle it (https://www.notion.so/eppo/Concatenated-experiment-key-design-d212aaaf42e7440bbdafd13cd0069dc8) and Agam updated the Java PR.

Reply:
I mention this in Agam's PR: https://github.com/Eppo-exp/java-server-sdk/pull/20#discussion_r1316339431

Re: the `null` allocation, I don't think we need to handle it. I've changed my mind after looking at the code more. 

If a matching rule is found, it should have an allocationKey (the allocation key is a required attribute in the rule dto https://github.com/Eppo-exp/eppo/blob/127e5da05cfb391f280a7304a75b77f1d1f7624c/packages/api/src/randomized-assignment/dto/randomized-assignment-targeting-rule.dto.ts#L5-L6), which will be logged here. 

If a matching rule is not found, then we early return and never reach this assignment logging function call (where we're trying to figure out how to handle a missing allocation key ..).